chromium_src/components/search_engines/android/template_url_service_android.cc

```diff
@@ -20,3 +23,59 @@ jboolean TemplateUrlServiceAndroid::DoesDefaultSearchEngineHaveLogo(
 
   return DoesDefaultSearchEngineHaveLogo_ChromiumImpl(env, obj);
 }
+
+jboolean TemplateUrlServiceAndroid::AddSearchEngine(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jstring>& search_engine_title,
```

Comment:
you can pass the native pointer from `getNativePtr`

Reply:
oic, you can't create one in java, nvm

Reply:
yeah correct. i was going to message that.